### ISSUE TITLE
How to exclusively use FCM

### ISSUE THREAD

**Issue by vijayb** [phonegap/phonegap-plugin-push#2012]
Hi guys, when using the push plugin on iOS if I omit the Google services plist file from the cordova project, when I deploy the app to my phone I see a registration call back that returns something like this:

{"registrationId":"9140acb85b865d377263523560f4bc20ddf1b99bbe78fe30bb8d4e6f9efa41b5","registrationType":"APNS"}

When I put the plist file in the project I see something like this

{"registrationId":"dbqA0KA9xTc:APA91bEXcdku6qAuuHUO98SMsqPnCOiheZzn66cmPHh9YBE1TqCnMWrREGrjZYihoqO4IS6z2Nn-aIF2dIJqjR8ajTpVqfnz4pxM7sjxOflaYxlq-Hgpn7V6ZDwVn67wEJH3HylLlavh":"FCM"}

I'm able to send push notifications to the app using Pusher and pushtry.com when using APNS, but cant' seem to get notifications working with the firebase console. I have configured the Firebase console with all the app information but I can't seem to get it to send a notification to the phone. I'm wondering if Firebase is sending to the production APNS server or the sandbox. But I don't seem to have a way to configure that and I'm not sure where to look.

Can anyone point me to something that will show me how to send notifications using FCM on iOS? I could just go with APNs, but it would much nicer if I could use FCM for both Android and iOS. Help appreciated!

**Comment by macdonst** [comment]
@vijayb seems to be the production/development problem. You could use some of the 
node-fcm scripts in the payload section of the docs to send test messages.

**Comment by vijayb** [comment]
Hi Simon, the part I find confusing is that when I don't use the plist file and it defaults to using APNS, I'm pretty sure I'm deploying to my phone using the production certificate. The difference is that to send the actual push notification you have to use the sandbox APNs. pushtry.com let's you specify to use the sandbox with the production checkbox. If you check production it will go to apps deployed on the app store. If you don't check it it will go to apps using the production certificate deployed directly to an iphone using Xcode.

The part I don't understand is how to do this with FCM. It seems that if FCM is being used, then it must know how to send to the underlying APNs server and decide whether to send to the production server or to the sandbox.

Do you guys have any experience using FCM to send push notifications to apps deployed to an iphone with Xcode? I feel like there is some simple checkbox I'm missing somewhere but for the life of me I can't figure it out after much googling

**Comment by vijayb** [comment]
This seems pretty relevant:
https://medium.com/developermind/using-firebase-cloud-messaging-for-remote-notifications-in-ios-d35de1dc67b2

It mentions:

"Important thing in here is FIRInstanceIDAPNSTokenType. There are three types available .prod, .sandbox and .none. To get notification, one of the Production or Sanbox token types must be used."

**Comment by macdonst** [comment]
@vijayb when you look at your entitlements file if the APS Environment is set to "development"

![screenshot 2017-10-16 15 40 54](https://user-images.githubusercontent.com/353180/31631689-70475434-b288-11e7-8637-c45548db3503.png)

Then you need to set `fcmSandbox: true` so that it goes to the development push service for a registration ID. Then you should send to the sandbox via pushtry.com.

If it is set to "production" then use `fcmSandbox: false` then you should send to the production via pushtry.com.

**Comment by vijayb** [comment]
Hi Simon, when I change the APS Environment string to "production" in Xcode, it then shows an error status in the capability:

![image](https://user-images.githubusercontent.com/957626/31690326-41b99838-b347-11e7-9dc2-7f79435f5bef.png)

**Comment by vijayb** [comment]
Also the other thing I'm confused about is when I setup the iOS app under the firebase console it never asks me to upload a p12 certificate at any point. So I'm a bit flummoxed how it can even have credentials to send notifications on APNs

**Comment by macdonst** [comment]
@vijayb right, so you need to do exactly what it says and add the aps-environment to the entitlements file. You can probably just click the "fix issue" button.

**Comment by vijayb** [comment]
Thanks Simon. Ok so this is a little embarrassing, but I think I fixed the issue by uploading the .p12 file to the firebase console. It was definitely bone-headed of me to not have thought about that earlier but it's kind of hidden away. Of course Push notifications from FCM to APNs aren't going to work if FCM doesn't have the correct APNs credentials. I found Google's docs on this pretty poor, unfortunately.

Anyway, so now I can use FCM to send push notifications to an iOS device deployed from xcode (huzzah!). I'm going to deploy to the app store now and hopefully getting notifications on the production release won't be too much harder than uploading the correct APNs .p12 file to the Firebase console. 

Thanks for your patience and help
